### PR TITLE
refactor: added the enhancements to ff4j-features

### LINKED ISSUE #276: ff4j-features micro-frontend improvements
- [x] linting of micro-frontend
```json
"scripts" {
  "lint:html:ff4j-features": "htmlhint --config lint/.htmlhintrc projects/ff4j-features/src/**/*.html",
  "lint:styles:ff4j-features": "stylelint --config lint/.stylelintrc.json projects/ff4j-features/**/*.scss",
  "lint:all": "run-p lint lint:html lint:styles lint:html:ff4j-features lint:styles:ff4j-features"
}
```

- [x] testing of micro-frontend 
```json
"scripts": {
    "test": "ng test --watch=false",
    "test:ff4j-features": "ng test ff4j-features --watch=false",
    "test:all": "run-s test test:ff4j-features",
}
```

- [x] coverage missing for micro-frontend. Turn on coverage option

```json
"test": {
   "builder": "@angular-devkit/build-angular:karma",
    "options": {
          "codeCoverage": true
     }
}
```

- [x] modules must be lazily loaded in the root route
```javascript
{
    path: 'features',
    loadChildren: () => import('../../projects/ff4j-features/src/app/FeaturesModule').then(m => m.FeaturesModule)
  }
```

- [x] no `entryComponents` in the module

- [x] Need to have two modules `FeaturesModule` and `FeaturesBrowserModule`. `FeaturesModule` will be used by the parent `AppModule` whereas `FeaturesBrowserModule` will be used in the micro-frontend i.e. `main.ts`

**FeaturesModule.ts**
```typescript
@NgModule({
  declarations: [FeaturesComponent],
  imports: [CommonModule, FeaturesRoutingModule],
  exports: [FeaturesComponent]
})
export class FeaturesModule {}
```

- [x] micro-frontend should have its own child routes

```javascript
const routes: Routes = [{ path: 'features', component: FeaturesComponent }];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class FeaturesRoutingModule {}
```

**FeaturesBrowserModule.ts**
```typescript
@NgModule({
  imports: [BrowserModule, FeaturesModule, RouterModule.forRoot([])],
  bootstrap: environment.loadBootstrap ? [FeaturesComponent] : []
})
export class FeaturesBrowserModule {
  constructor(private injector: Injector) {}

  ngDoBootstrap() {
    const ff4jFeaturesElement = createCustomElement(FeaturesComponent, {injector: this.injector});
    customElements.define('ff4j-features-element', ff4jFeaturesElement);
  }
}
```

- [x] micro-frontend `styles` to refer `src\styles` (change has to be made in **angular.json**)

### PR DESCRIPTION
**Pull request by J-DK**
# Description

Added the enhancement needed for the features micro front end

Fixes #276 

## Type of change

Please check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -133,7 +133,7 @@
       },
       "root": "projects/ff4j-features",
       "sourceRoot": "projects/ff4j-features/src",
-      "prefix": "ff4j-features",
+      "prefix": "ff4j",
```
Comment (paul58914080) [inline angular.json]: nice 👍

File: package.json [hunk 2]
```diff
@@ -26,14 +26,17 @@
     "build:features-element": "ng build ff4j-features --prod --output-hashing none && node build-elements/ff4j-features.js",
     "build:properties-element": "ng build ff4j-properties --prod --output-hashing none && node build-elements/ff4j-properties.js",
     "build:dashboard-element": "ng build ff4j-dashboard --prod --output-hashing none && node build-elements/ff4j-dashboard.js",
-    "test": "ng test",
-    "test:once": "ng test --watch=false",
+    "test": "ng test --watch=false",
```
Comment (paul58914080) [inline package.json]: Thank you for this 👍

File: projects/ff4j-features/src/app/features-routing.module.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { NgModule } from '@angular/core';
+import { FeaturesComponent } from './features.component';
+import { RouterModule, Routes } from '@angular/router';
+
+
+const routes: Routes = [{ path: '', component: FeaturesComponent }];
```
Comment (paul58914080) [inline projects/ff4j-features/src/app/features-routing.module.ts]: path should have features

Comment (J-DK) [reply projects/ff4j-features/src/app/features-routing.module.ts]: I don't think so Paul, if **features** is there in the path, then from the main app, we have to reference it like `'4200/features/features'`

Comment (paul58914080) [reply projects/ff4j-features/src/app/features-routing.module.ts]: I get your point 👍

**Comment by paul58914080** [comment]
thank you so much @J-DK for your contribution.

**Pull request merged**